libfwupdplugin/fu-common.c

```diff
@@ -2069,3 +2070,196 @@ fu_common_is_cpu_intel (void)
 #endif
 	return FALSE;
 }
+
+#define UDISKS_DBUS_SERVICE		"org.freedesktop.UDisks2"
+#define UDISKS_DBUS_PATH		"/org/freedesktop/UDisks2/Manager"
+#define UDISKS_DBUS_MANAGER_INTERFACE	"org.freedesktop.UDisks2.Manager"
+#define UDISKS_DBUS_PART_INTERFACE 	"org.freedesktop.UDisks2.Partition"
+#define UDISKS_DBUS_FILE_INTERFACE	"org.freedesktop.UDisks2.Filesystem"
+
+static GPtrArray *
+fu_common_get_block_devices (GDBusConnection *connection, GError **error)
+{
+	GVariantBuilder builder;
+	GVariant *input;
+	const gchar *obj;
+	g_autoptr(GVariant) output = NULL;
+	g_autoptr(GDBusProxy) proxy = NULL;
+	g_autoptr(GPtrArray) devices = NULL;
+	g_autoptr(GVariantIter) iter = NULL;
+
+	proxy = g_dbus_proxy_new_sync (connection,
+				       G_DBUS_PROXY_FLAGS_NONE, NULL,
+				       UDISKS_DBUS_SERVICE,
+				       UDISKS_DBUS_PATH,
+				       UDISKS_DBUS_MANAGER_INTERFACE,
+				       NULL, error);
+	if (proxy == NULL) {
+		g_prefix_error (error, "failed to find %s: ", UDISKS_DBUS_SERVICE);
+		return NULL;
+	}
+	g_variant_builder_init (&builder, G_VARIANT_TYPE_VARDICT);
+	input = g_variant_new ("(a{sv})", &builder);
+	output =  g_dbus_proxy_call_sync (proxy,
+					  "GetBlockDevices", g_variant_ref (input),
+					  G_DBUS_CALL_FLAGS_NONE,
+					  -1, NULL, error);
+	if (output == NULL)
+		return NULL;
+	devices = g_ptr_array_new_with_free_func (g_free);
+	g_variant_get (output, "(ao)", &iter);
+	while (g_variant_iter_next (iter, "o", &obj))
+		g_ptr_array_add (devices, g_strdup (obj));
+
+	return g_steal_pointer (&devices);
+}
+
+/**
+ * fu_common_get_volumes_by_kind:
+ * @kind: A volume kind, typically a GUID
+ * @error: A #GError or NULL
+ *
+ * Call into the plugin's get results routine
+ *
+ * Finds all volumes of a specific type
+ *
+ * Returns: (transfer container) (element-type FuVolume): a #GPtrArray, or %NULL if the kind was not found
+ *
+ * Since: 1.5.0
```

Comment:
Not going to be a backport to 1.4.6 feature?

Reply:
I was tempted to push 1.5.0 soon. I can do these for 1.4.6 if you'd prefer.

Reply:
Well I'm just hesitant on 1.5.0 until there is more !intel in HSI..

Reply:
Fair point. Moved to 1.4.6